### PR TITLE
trino-cli: init at 387

### PR DESCRIPTION
**Pull request by regadas**
###### Description of changes

Add trino-cli. The Trino CLI provides a terminal-based, interactive shell for running queries

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/database/trino-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    meta = with lib; {
+      description =
+        "The Trino CLI provides a terminal-based, interactive shell for running queries";
+      homepage = "https://github.com/trinodb/trino";
+      license = licenses.asl20;
+      maintainers = [ maintainers.regadas ];
+    };
+
+  };
+in this
```
Comment (legendofmiracles) [inline pkgs/development/tools/database/trino-cli/default.nix:39]:
```suggestion
```

File: pkgs/development/tools/database/trino-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ callPackage, lib, stdenv, fetchurl, jre, makeWrapper }:
+
+let
+  this = stdenv.mkDerivation rec {
```
Comment (legendofmiracles) [inline pkgs/development/tools/database/trino-cli/default.nix:4]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/tools/database/trino-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    version = "387";
+    pname = "trino-cli";
```
Comment (legendofmiracles) [inline pkgs/development/tools/database/trino-cli/default.nix:6]:
```suggestion
  pname = "trino-cli";
  version = "387";
```

File: pkgs/development/tools/database/trino-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+      makeWrapper ${jre}/bin/java $out/bin/trino \
+        --add-flags "-jar $out/share/java/${jarfilename}"
+
+      runHook postInstall
+    '';
+
+    meta = with lib; {
+      description =
+        "The Trino CLI provides a terminal-based, interactive shell for running queries";
```
Comment (legendofmiracles) [inline pkgs/development/tools/database/trino-cli/default.nix:32]:
```suggestion
      description = "Terminal-based, interactive shell for running queries";
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -35630,4 +35630,7 @@ with pkgs;
   honeyvent = callPackage ../servers/tracing/honeycomb/honeyvent { };
 
   mictray = callPackage ../tools/audio/mictray { };
+
+  trino-cli = callPackage ../development/tools/database/trino-cli { };
```
Comment (legendofmiracles) [inline pkgs/top-level/all-packages.nix:35634]: Please don't put new entries at the bottom of the file. You can grep for the first couple of letters of trino to see where it should belong. Ideally alphabetically sorted

File: pkgs/development/tools/database/trino-cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
+{ callPackage, lib, stdenv, fetchurl, jre, makeWrapper }:
```
Comment (legendofmiracles) [inline pkgs/development/tools/database/trino-cli/default.nix:1]:
```suggestion
{ lib, stdenv, fetchurl, jre, makeWrapper }:
```

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

**Comment by regadas** [comment]
@legendofmiracles thanks for the review!

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (legendofmiracles) [reply pkgs/top-level/all-packages.nix:35634]: Yeah, it's messy, but we can't fix it without breaking all PRs touching that file.

In the meantime putting them all at the bottom of the file is the worst option. 
There's also another PR that was opened today that did. 
If we merged either, the other one would have a merge conflict. 
Better to disperse it

**Comment by regadas** [comment]
@legendofmiracles I believe I addressed all of your comments. Please take a look when you get a chance. Thanks!

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -35630,4 +35632,5 @@ with pkgs;
   honeyvent = callPackage ../servers/tracing/honeycomb/honeyvent { };
 
   mictray = callPackage ../tools/audio/mictray { };
+
```
Comment (legendofmiracles) [inline pkgs/top-level/all-packages.nix:35635]:
```suggestion
```

File: pkgs/development/tools/database/trino-cli/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, fetchurl, jre, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  version = "387";
```
Comment (legendofmiracles) [inline pkgs/development/tools/database/trino-cli/default.nix:4]: pname goes first, and after that is version

Comment (regadas) [reply pkgs/development/tools/database/trino-cli/default.nix:4]: gotcha 👍 , are these guidelines documented somewhere?

**Comment by regadas** [comment]
@legendofmiracles comments addressed, thanks!

File: maintainers/maintainer-list.nix [hunk 9]
```diff
@@ -14675,4 +14675,10 @@
     github = "npatsakula";
     githubId = 23001619;
   };
+  regadas = {
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:14678]: Should also not be at the bottom of the file

**Review by legendofmiracles — COMMENTED** [review]
Please squash the commits

**Comment by regadas** [comment]
@legendofmiracles done.

**Pull request merged**